caffe2/python/test/gpu_context_test.py

```diff
@@ -14,7 +14,9 @@
                  "THC pool testing is obscure and doesn't work on HIP yet")
```

Comment:
so my previous PR actually just changed this test to not run on HIP. This combination of THCAllocator + Caffe2 is really obscure and I don't think it's important to support it officially. So feel free to abandon you PR. Thanks a lot for jumping with the fix and sorry again for glancing over the CI in the original PR.

Reply:
I'm good with this PR still landing.  We made an honest attempt at getting this feature to work, even if the test is ultimately skipped.

Reply:
Sounds good, I'm landing it.